.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,285 @@
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle  # This enables Gradle caching
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Check Gradle build scripts dependencies are sorted
+        run: ./gradlew buildEnvironment --dry-run
+
+      - name: Find affected modules
+        id: affected-modules
+        run: |
+          # Determine the base commit to compare against
+          BASE_BRANCH="origin/${{ github.base_ref }}"
+
+          echo "Comparing against base: $BASE_BRANCH"
+
+          CHANGED_FILES=$(git diff --name-only $BASE_BRANCH...HEAD)
+
+          # Create a list of affected modules
+          AFFECTED_MODULES=""
+
+          # Check for changes in each module directory
+          for dir in core android integrations/*; do
+            if [ -d "$dir" ] && echo "$CHANGED_FILES" | grep -q "^$dir/"; then
+              MODULE_PATH=$(echo $dir | sed 's/\//:/')
+              if [ -z "$AFFECTED_MODULES" ]; then
+                AFFECTED_MODULES=":$MODULE_PATH"
+              else
+                AFFECTED_MODULES="$AFFECTED_MODULES,:$MODULE_PATH"
+              fi
+              echo "$dir=true" >> $GITHUB_OUTPUT
+            else
+              if [ -d "$dir" ]; then
+                echo "$dir=false" >> $GITHUB_OUTPUT
+              fi
+            fi
+          done
+
+          echo "affected_modules=$AFFECTED_MODULES" >> $GITHUB_OUTPUT
+          echo "Affected modules: $AFFECTED_MODULES"
+
+      - name: Run Detekt on affected modules
+        run: |
+          AFFECTED_MODULES="${{ steps.affected-modules.outputs.affected_modules }}"
+          if [ -n "$AFFECTED_MODULES" ]; then
+            IFS=',' read -ra MODULES <<< "$AFFECTED_MODULES"
+            for module in "${MODULES[@]}"; do
+              echo "Running detekt for $module"
+              ./gradlew ${module}:detekt
+            done
+          else
+            echo "No modules affected, skipping detekt"
+          fi
+
+      - name: Run Android Lint on affected modules
+        run: |
+          AFFECTED_MODULES="${{ steps.affected-modules.outputs.affected_modules }}"
+          if [ -n "$AFFECTED_MODULES" ]; then
+            IFS=',' read -ra MODULES <<< "$AFFECTED_MODULES"
+            for module in "${MODULES[@]}"; do
+              # Skip core module for lint
+              if [[ "$module" != ":core" ]]; then
+                echo "Running lint for $module"
+                ./gradlew ${module}:lint
+              fi
+            done
+          else
+            echo "No modules affected, skipping lint"
+          fi
+
+      - name: Run SonarCloud Analysis
+        uses: SonarSource/sonarqube-scan-action@2500896589ef8f7247069a56136f8dc177c27ccf  # v5.2.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```

Comment:
This is being run on the whole PR (instead of only affected modules). We can consider creating a parallel job for this in any upcoming PR.